src/lib.rs

```diff
@@ -802,6 +801,138 @@ impl<M, R, S> Client<M, R, S> {
             async move { Ok(Request::Local(tx)) }
         }
     }
+
+    /// Performs a request for which the server returns a oneshot receiver.
+    pub async fn rpc<Req, Res>(&self, msg: Req) -> Result<Res, Error>
+    where
+        S: Service,
+        M: From<WithChannels<Req, S>> + Send + Sync + Unpin + 'static,
+        R: From<Req> + Serialize + 'static,
+        Req: Channels<S, Tx = channel::oneshot::Sender<Res>, Rx = NoReceiver>,
+        Res: RpcMessage,
+    {
+        let recv: channel::oneshot::Receiver<Res> = match self.request().await? {
+            Request::Local(request) => {
+                let (tx, rx) = channel::oneshot::channel();
+                request.send((msg, tx)).await?;
+                rx
+            }
+            #[cfg(not(feature = "rpc"))]
+            Request::Remote(_request) => unreachable!(),
+            #[cfg(feature = "rpc")]
+            Request::Remote(request) => {
+                let (_tx, rx) = request.write(msg).await?;
+                rx.into()
+            }
+        };
+        let res = recv.await?;
+        Ok(res)
+    }
+
+    /// Performs a request for which the server returns a spsc receiver.
+    pub async fn server_streaming<Req, Res>(
+        &self,
+        msg: Req,
+        local_response_cap: usize,
+    ) -> Result<channel::spsc::Receiver<Res>, Error>
+    where
+        S: Service,
+        M: From<WithChannels<Req, S>> + Send + Sync + Unpin + 'static,
```

Comment:
Maybe not now, but we should probably DRY all these Send + Sync + 'static bounds so we can change them in a single place for WASM...

Reply:
FWIW, I'm considering adding [`CondSend` and `CondSync`](https://github.com/wnfs-wg/rs-wnfs/blob/main/wnfs-common/src/utils/send_sync_poly.rs#L31-L61) to `n0-future`.

But also: As long as messages don't have to hold on to references to browser stuff (such as `JsValue` or similar), it's totally fine to require them to be `Send + Sync`.